### PR TITLE
Support autocorrect for `Style/CaseEquality` cop

### PR DESCRIPTION
**Pull request by fatkodima**

### CODE REVIEW

**Comment by tejasbubane** [comment]
@fatkodima Since rubocop 0.88 is released, you should update `VersionChanged` & move changelog under `master`.

**Comment by fatkodima** [comment]
Updated.

File: lib/rubocop/cop/style/case_equality.rb [hunk 1]
```diff
@@ -29,13 +29,20 @@ module Style
       #   (1..100).include?(7)
       #   some_string =~ /something/
       #
-      class CaseEquality < Cop
+      class CaseEquality < Base
```
Comment (tejasbubane) [inline lib/rubocop/cop/style/case_equality.rb:32]: I think this should not be changed:

```suggestion
      class CaseEquality < Cop
```

Comment (koic) [reply lib/rubocop/cop/style/case_equality.rb:32]: The auto-correction has been added by the new API, I think it is okay to extend `Base`. Rather, changing the inheritance to `Cop` will fail tests.

Comment (tejasbubane) [reply lib/rubocop/cop/style/case_equality.rb:32]: @koic I might be a little confused between Base and Cop. I see most cops inherit from `Cop`, do not `extend AutoCorrector` and have separate method `autocorrect`. Which of these is the preferred/new approach?

Comment (marcandre) [reply lib/rubocop/cop/style/case_equality.rb:32]: Inheriting from  `Base` is preferred/new approach. Here's the doc: https://docs.rubocop.org/rubocop/v1_upgrade_notes.html

File: lib/rubocop/cop/style/case_equality.rb [hunk 2]
```diff
@@ -47,6 +54,18 @@ def const?(node)
...
+
+        def replacement(lhs, rhs)
+          case lhs.type
+          when :regexp
+            "#{rhs.source} =~ #{lhs.source}"
+          when :begin
+            child = lhs.children.first
+            "#{lhs.source}.include?(#{rhs.source})" if child&.range_type?
+          when :const
+            "#{rhs.source}.is_a?(#{lhs.source})"
```
Comment (knu) [inline lib/rubocop/cop/style/case_equality.rb:66]: Not all constants are class/module instances.  Regexps are often bound to a constant, to be used with `===`.

I think a better bet is to replace the constant only if it is CamelCased.

File: lib/rubocop/cop/style/case_equality.rb [hunk 3]
```diff
@@ -47,6 +54,18 @@ def const?(node)
             true
           end
         end
+
+        def replacement(lhs, rhs)
+          case lhs.type
+          when :regexp
+            "#{rhs.source} =~ #{lhs.source}"
```
Comment (knu) [inline lib/rubocop/cop/style/case_equality.rb:61]: `/regexp/ === object` and `object =~ /regexp/` are not interchangeable.  The former returns true or false whereas the latter returns an index or nil.

File: lib/rubocop/cop/style/case_equality.rb (hunk 2 above)
Comment (koic) [reply lib/rubocop/cop/style/case_equality.rb:66]: Thank you for your feedback. I've opened the PR #8526.

File: lib/rubocop/cop/style/case_equality.rb (hunk 3 above)
Comment (koic) [reply lib/rubocop/cop/style/case_equality.rb:61]: Exactly! I've opened the PR #8527.

**Pull request merged**